src/core/algorithms/ind/mind/mind.h

```diff
@@ -44,7 +45,13 @@ class Mind final : public INDAlgorithm {
     bool SetExternalOption(std::string_view option_name, boost::any const& value) override;
     void LoadINDAlgorithmDataInternal() override;
 
-    bool TestCandidate(RawIND const& raw_ind);
+    ///
+    /// Test a given IND candidate to determine if it should be registered.
+    ///
+    /// \return `false` with `std::nullopt` if the candidate should not be registered,
+    ///         otherwise return `true` along with the associated error threshold.
+    ///
+    std::pair<bool, std::optional<config::ErrorType>> TestCandidate(RawIND const& raw_ind);
```

Comment:
So `false` always means no value, and `true` means that the value will be present. I feel like we can simplify this logic by simply utilizing `std::optional<config::ErrorType>`, the interface will be much simplier